File: agent/app/model/windows_service.go
package model

type WindowsService struct {
	BaseModel
	Name            string `json:"name" gorm:"type:varchar(128);not null;uniqueIndex"`
	DisplayName     string `json:"displayName" gorm:"type:varchar(128);not null"`
	ServiceType     string `json:"serviceType" gorm:"type:varchar(64);not null"`
	ExecPath        string `json:"execPath" gorm:"type:varchar(512);not null"`
	Args            string `json:"args" gorm:"type:text"`
	WorkDir         string `json:"workDir" gorm:"type:varchar(512)"`
	EnvFilePath     string `json:"envFilePath" gorm:"type:varchar(512)"`
	DLLDir          string `json:"dllDir" gorm:"type:varchar(512)"`
	ConfigPath      string `json:"configPath" gorm:"type:varchar(512)"`
	JarPath         string `json:"jarPath" gorm:"type:varchar(512)"`
	WinSWPath       string `json:"winswPath" gorm:"type:varchar(512)"`
	ServicePath     string `json:"servicePath" gorm:"type:varchar(512)"`
	Status          string `json:"status" gorm:"type:varchar(64)"`
	Message         string `json:"message" gorm:"type:text"`
	RegisterService bool   `json:"registerService"`
	AutoStart       bool   `json:"autoStart"`
}

File: frontend/src/api/interface/windows-service.ts
export namespace WindowsService {
    export interface ConfigTemplateField {
        envKey: string;
        defaultValue: string;
        value: string;
        configPath: string;
        inputType: 'text' | 'number' | 'password';
        sensitive: boolean;
    }

    export interface ConfigTemplate {
        enabled: boolean;
        template: string;
        fileName?: string;
        generatedPath?: string;
        content?: string;
        dynamicFields: ConfigTemplateField[];
    }

    export interface Item {
        id: number;
        name: string;
        displayName: string;
        serviceType: 'java' | 'dll' | 'package';
        execPath: string;
        args: string;
        workDir: string;
        envFilePath: string;
        dllDir: string;
        configPath: string;
        jarPath: string;
        winswPath: string;
        servicePath: string;
        uninstallScriptPath: string;
        status: string;
        message: string;
        configTemplate?: ConfigTemplate | null;
        registerService: boolean;
        autoStart: boolean;
    }

    export interface Create {
        name?: string;
        displayName?: string;
        serviceType: 'java' | 'dll' | 'package';
        execPath?: string;
        args?: string;
        workDir: string;
        envFilePath?: string;
        dllDir?: string;
        configPath?: string;
        jarPath?: string;
        winswPath?: string;
        configTemplate: ConfigTemplate | null;
        registerService: boolean;
        autoStart: boolean;
    }

    export interface UploadServicePreview {
        name: string;
        displayName: string;
        serviceType: 'java' | 'dll' | 'package';
        runtimeType: string;
        workDir: string;
        configPath: string;
        jarPath: string;
        execPath: string;
        args: string;
        winswPath: string;
        sourcePath: string;
        composeServiceName: string;
    }

    export interface UploadJarResult {
        jarPath: string;
        workDir: string;
        execPath: string;
        winswPath: string;
        fileName: string;
        installDir: string;
        configTemplate?: ConfigTemplate | null;
        services?: UploadServicePreview[];
    }

    export interface Update extends Create {
        id: number;
    }

    export interface Operate {
        id: number;
        operate: 'start' | 'stop' | 'restart' | 'enable' | 'disable' | 'status';
    }

    export interface ConfigFile {
        type: 'config' | 'env';
        path: string;
        content: string;
    }

    export interface ConfigFileUpdate {
        type: 'config' | 'env';
        content: string;
    }

    export interface LogFile {
        name: string;
        path: string;
        exists: boolean;
        size: number;
        modifiedAt: string;
    }
}
